File: backEnd/src/main/java/com/bambooByte/horaDoBicho/entities/Notificacao.java
package com.bambooByte.horaDoBicho.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Notificacao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idNotificacao;
    @Column(nullable = false)
    private Long idCliente;
    @Column(nullable = true)
    private Long idAgendamento;
    @Column(nullable = false)
    private String titulo;
    @Column(nullable = false)
    private String descricao;
    @Column(nullable = false)
    private boolean lida;

    public Notificacao() {
    }

    public Notificacao(Long idCliente, String titulo, String descricao, Long idAgendamento) {
        this.idCliente = idCliente;
        this.titulo = titulo;
        this.descricao = descricao;
        this.idAgendamento = idAgendamento;
        this.lida = false;
    }
        public Long getIdAgendamento() {
            return idAgendamento;
        }

        public void setIdAgendamento(Long idAgendamento) {
            this.idAgendamento = idAgendamento;
        }
    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public Long getIdNotificacao() {
        return idNotificacao;
    }

    public void setIdNotificacao(Long idNotificacao) {
        this.idNotificacao = idNotificacao;
    }

    public Long getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(Long idCliente) {
        this.idCliente = idCliente;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public boolean isLida() {
        return lida;
    }

    public void setLida(boolean lida) {
        this.lida = lida;
    }

}

File: hora_do_bicho/lib/models/notificacao.dart
class Notificacao {
  final int idNotificacao;
  final int idCliente;
  final String titulo;    
  final String descricao;  
  final bool lida; 
  final int idAgendamento;        

  Notificacao({
    required this.idNotificacao,
    required this.idCliente,
    required this.titulo,
    required this.descricao,
    required this.lida,
    required this.idAgendamento
  });

  factory Notificacao.fromJson(Map<String, dynamic> json) {
    return Notificacao(
      idNotificacao: json['idNotificacao'],
      idCliente: json['idCliente'],
      titulo: json['titulo'],
      descricao: json['descricao'],
      lida: json['lida'] ?? false,
      idAgendamento: json['idAgendamento']
    );
  }
}
